hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/cache/CacheAddInvalidationListenerTask.java

```diff
@@ -46,13 +51,24 @@ public void handleEvent(Object eventObject) {
                     return;
                 }
                 if (eventObject instanceof CacheInvalidationMessage) {
-                    CacheInvalidationMessage message = (CacheInvalidationMessage) eventObject;
-
-                    ClientMessage eventMessage
-                            = CacheAddInvalidationListenerCodec.encodeCacheInvalidationEvent(message.getName(),
-                            message.getKey(), message.getSourceUuid());
-                    sendClientMessage(message.getKey(), eventMessage);
-
+                    if (eventObject instanceof CacheSingleInvalidationMessage) {
+                        CacheSingleInvalidationMessage message = (CacheSingleInvalidationMessage) eventObject;
+                        ClientMessage eventMessage = CacheAddInvalidationListenerCodec.
+                                    encodeCacheInvalidationEvent(message.getName(), message.getKey(), message.getSourceUuid());
+                        sendClientMessage(message.getName(), eventMessage);
```

Comment:
I see that you changed the orderKey from key to distributed object name. Even Though it does not break the current guarantee (in fact, it causes more strict guarantee), I was wondering why did you decided to change that?